### ISSUE TITLE
"buffer was not aligned for 64-bit numbers" under miri

### ISSUE THREAD

**Issue by alecmocatta** [shepmaster/twox-hash#37]
```rust
let mut hasher = twox_hash::XxHash::with_seed(0);
hasher.write_u64(0);
let _ = hasher.finish();
```

hits this debug_assert under miri:
https://github.com/shepmaster/twox-hash/blob/ea71112ef37fede6ac538abcdd181f69bcf21592/src/sixty_four.rs#L160

resulting in the following backtrace:
```
error[E0080]: Miri evaluation error: the evaluated program panicked at 'buffer was not aligned for 64-bit numbers', /Users/alecmocatta/.cargo/registry/src/github.com-1ecc6299db9ec823/twox-hash-1.4.2/src/sixty_four.rs:160:9
  |
  = note: inside call to `twox_hash::sixty_four::Buffer::as_u64s_and_u32s` at /Users/alecmocatta/.cargo/registry/src/github.com-1ecc6299db9ec823/twox-hash-1.4.2/src/sixty_four.rs:235:60
  = note: inside call to `twox_hash::sixty_four::XxHash64::finish` at /Users/alecmocatta/.cargo/registry/src/github.com-1ecc6299db9ec823/twox-hash-1.4.2/src/sixty_four.rs:285:9
  = note: inside call to `<twox_hash::sixty_four::XxHash64 as std::hash::Hasher>::finish` at /Users/alecmocatta/Documents/deploy/build_id/src/lib.rs:119:80
```

Adding more calls to `hasher.write_u64(0);` it also hits this debug_assert: https://github.com/shepmaster/twox-hash/blob/ea71112ef37fede6ac538abcdd181f69bcf21592/src/sixty_four.rs#L146

**Comment by shepmaster** [comment]
Sigh. Thank you!

Miri adheres to the letter of the law:

> `align_to` in Miri always puts everything into the first component [... exploiting ...] the freedom that the `align_to` spec grants
>
> The spec says:
>
> > The method does a best effort to make the middle slice the greatest length possible for a given type and input slice, but only your algorithm's performance should depend on that, not its correctness.

I'll probably wrap this as part of #34 and discard usage of `align_to` completely.

**Comment by shepmaster** [comment]
Closed via #34
